Add file_random_recommender and update __init__.py

clementcanel opened the pull request.

rburke2233 committed:
Initial commit

AnasBuhayh committed:
Add files via upload

AnasBuhayh committed:
Add files via upload

clementcanel committed:
Test and added null option to simulated_response method. Simple logic if user doesn't click system returns null option=True

clementcanel committed:
Implemented logic for null option in choice_model. Added print statements to track document probabilities, consumer preferences, and selection reasoning.

clementcanel committed:
Modified _score_documents to apply the threshold on probabilities instead of scores, allowing users to opt out (null option) when no documents are sufficiently relevant

clementcanel committed:
working null

clementcanel committed:
Help with FileNotFoundError

clementcanel committed:
combined movies.csv with moves_metadata.csv on titles and years to associate movies with the production company

clementcanel committed:
rating_with_providers.csv

clementcanel committed:
9% unknown

AnasBuhayh committed:
updated experiments to include null-option

AnasBuhayh committed:
Merge pull request #1 from that-recsys-lab/null-option

Null option

AnasBuhayh committed:
installing surprise

AnasBuhayh committed:
surprise implementation

AnasBuhayh committed:
Merge pull request #2 from that-recsys-lab/surprise-recs

Surprise recs

AnasBuhayh committed:
updated requirements

AnasBuhayh committed:
Merge pull request #3 from that-recsys-lab/surprise-recs

updated requirements

AnasBuhayh committed:
restructure

AnasBuhayh committed:
more refactoring

AnasBuhayh committed:
refatoring code

AnasBuhayh committed:
kl-divergence

AnasBuhayh committed:
smores 2.0

AnasBuhayh committed:
Merge pull request #4 from that-recsys-lab/restruc

SMORES 2.0

AnasBuhayh committed:
Delete structure.md

AnasBuhayh committed:
Update .gitignore

AnasBuhayh committed:
add data files

AnasBuhayh committed:
Merge pull request #5 from that-recsys-lab/restruc

add data files

AnasBuhayh committed:
Delete notebooks directory

AnasBuhayh committed:
Update README.md

AnasBuhayh committed:
Update README.md

AnasBuhayh committed:
Update README.md

clementcanel committed:
Updated requirements.txt for NumPy compatibility and fixed random.sample() in SVD.py

clementcanel committed:
Added config file and added some modularity to the choice logic in CategorySimilarityLogit, and consumer class now uses choice_model instance, so no need to modify consumer code any longer

rburke2233 committed:
Top level implementation

__main__.py, Smores and SmoresState as well as configuration. See doc/sample-config.yaml for the structure.

rburke2233 committed:
Beginning consumer and consumer utility model implementation

Also added test cases

clementcanel committed:
Added CategorySimilarityLogitModel for consumer item choice with some unit tests for model registration

rburke2233 committed:
Provider class

rburke2233 committed:
Merge branch 'refactoring' of https://github.com/that-recsys-lab/smores into refactoring

rburke2233 committed:
Switched terminology for selection and choice

Now using item_selection and recommender_choice to avoid ambiguity.

rburke2233 committed:
Implemented provider utility model and provider factor

Also some changes to consumer to allow for factory here, too.
Not 100% sure if the ProviderData object is a good idea.

rburke2233 committed:
Merge pull request #6 from that-recsys-lab/refactoring_provider

Implemented provider utility model and provider factor

rburke2233 committed:
Started implementation of LensKit recommenders

rburke2233 committed:
Initial implementation of the recommender class

Note: needs unit tests

rburke2233 committed:
Now with test cases

Fixes to interaction history implementation

clementcanel committed:
Add file_random_recommender and update __init__.py

rburke2233 commented:
A couple of changes to suggest:
* This recommender should have its own item file for cold start recommendations. For example, if it is the Horror recommender, it will be a list of the top horror movies, not the whole movie catalog. So, there should be a separate data file specific to the recommender and the file name should be specified in a parameter local to this agent's definition. For example, the configuration could be:
  - name: Generic
    class_name: item_knn_cold
    min_neighbors: 1
    max_neighbors: 20
    min_similarity: 0.001
    cold_start_fallback:
      class_name: file_random
      item_file: popular_generic.csv
    cold_user_fallback:
      class_name: most_popular

And then it would read items from the "popular_generic.csv" file. 

* I would like to avoid Pandas if we can. It is overkill for just a list of file ids. We can read it into a list and shuffle each time.

rburke2233 committed:
Configuration and other updates

File Random Recommender is still not working. See comment on PR. Also started work on triggers, which probably should be its own branch.

rburke2233 committed:
Minor fixes

rburke2233 committed:
Update .gitignore
